### DISCUSSION TITLE
How to generate initial condition positions with a coarse grained 10 bead polymer model?

### DISCUSSION THREAD

**Discussion by Learnian** [marrink-lab/polyply_1.0#375] [Q&A] [answered]
Hi there! 

I am trying to generate initial location conditions for 200 coarse grained generic polymers, each of length 10 beads. Each bead has a radius of 1, and they are to fit in a cubic box of side length 12.628. They are the kind of polymers described in Model C of the follwoing paper [https://chemrxiv.org/engage/chemrxiv/article-details/6647c97491aefa6ce14a4a92](https://chemrxiv.org/engage/chemrxiv/article-details/6647c97491aefa6ce14a4a92). 
I think I have to create my own polymer library for this, instead of using any existing polymer library? Is this correct? If so how is it done?

**Comment by fgrunewald** [comment, marked as answer]
Hi @Learnian, 

Indeed you would have to create your own ff-files, but they are rather simple (see below). To generate the system coordinates you'll need a very simple topology file as shown below. To generate your polymer itp file use `polyply gen_params -f ff_ile.ff -seq A:10 -o A.itp -name A`. Then use the top file to generate the coordinates with `polyply gen_coords -p system.top -box 12.628 12.628 12.628 -o coords.gro`. If you want to use the WCM model to bias the initial end-to-end distance you can provide a build file though I'm not sure it makes any sense for such short polymers. 

Also, be aware that GROMACS might not be able to run the model.  We're exploring other MD engines but haven't made much progress yet. 

**FF-File**
```
[ moleculetype ]
A  1
[ atoms ]
1 A      1    A   A      1 0.0 72.0

[ link ]
resname "A"
[ bonds ]
A   +A  1   1.0  8000
```

**TOP-File**
```
[ defaults ]
1 2 ; sigma-epsilon format of LJ parameters

[ atomtypes ]
D 72.0 0.000 A 1.0 1.0

#include "A.itp"

[ system ]
polymers of A
[ molecules ]
A 200
```

**Reply by Learnian** [reply]
Great thanks a ton! I used this, but it gave me a KeyError ["A"], so I changed the D to an A and it worked fine. Is this ok? 

Also , I was wondering if there was a way to change a random seed or something that would give me a different initial configuration file for the same system each time? 

Nevermind, it looks like polyply already makes it different each time you run gen_coords. 

Thanks

**Reply by fgrunewald** [reply]
@Learnian indeed there should be an A; that's a typo on my part. Coordinates are always generated with different random seeds.
